### PR TITLE
Add Inputs.form

### LINKED ISSUE #73: Compound (multi-value) inputs?
Something that I find myself doing again and again is to join multiple inputs into a single value object. The value object acts as a proxy to the individual inputs.

It would be great to have such a helper as part of the library.

Example ([demo](https://observablehq.com/d/f3bc3f9ea45b9f73)):
```javascript
function mergeInputs(inputs, target = {}) {
  const props = Object.fromEntries(Object.entries(inputs).map(([key, input]) => [key, {
    enumerable: true,
    configurable: true,
    get: () => input.value,
    set: value => input.value = value,
  }]));
  return Object.defineProperties(target, props);
}
```
```javascript
viewof opts = {
  const inputs = {
    n: Range([1, 30], {value: 19, step: 1, label: "size" }),
    a: Range([0, 45], {value: 27.5, step:.1, label: "angle" }),
    x: Range([-100, 100], {value: 0, step: 1, label: "x offset" }),
    y: Range([-100, 100], {value: 0, step: 1, label: "y offset" }),
  };
  const view = html`<div>${Object.values(inputs)}`;
  view.value = mergeInputs(inputs);
  return view;
}
```
```
// opts
Object {n: 19, a: 27.5, x: 0, y: 0}
```

### PR DESCRIPTION
**Pull request by mbostock**
Fixes #73. Demo: https://observablehq.com/d/4108c4771279ad1a

### CODE REVIEW

**Comment by mbostock** [comment]
Any thoughts on this one, @Fil? (I’ll get to that Legends PR soon…)

**Comment by Fil** [comment]
I like the fact that it's hierarchical, like the good old dat-gui; however in that case, we could want a way to pass styling or fieldset+label options to the children, if we want the inputs structure to be visible? This would be useful in general but particularly so for nested inputs. But maybe that's for another project.

Another idea is that we could also support a Map of inputs, that would return a Map of values (or maybe a key.value object). It would seem more consistent with Inputs.select, for example, and lists of options generated by, say, d3.rollup.

File: src/form.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import {html} from "htl";
+
+export function form(inputs) {
+  return (Array.isArray(inputs) ? arrayForm : objectForm)(inputs);
+}
+
+function arrayForm(inputs) {
+  inputs = [...inputs]; // defensive copy
+  let value; // lazily constructed
+  const form = html`<form>${inputs}`;
```
[the remark this answers is not in the document]
Comment (mbostock) [reply src/form.js]: Okay, switched to a div.

**Comment by mbostock** [comment]
> I'd love see an option for a template callback added that simply receives the inputs array/object as passed to Inputs.form(), and returns the outermost element.

Nice idea. Added.

**Review by Fil — APPROVED** [review]

**Comment by mootari** [comment]
When will the new release get deployed to Observable?

**Comment by mbostock** [comment]
In the near future.

**Comment by harrybiddle** [comment]
Is it possible for the properties of an input in a form to depend on the value of another in the same form?

**Pull request merged**